### PR TITLE
Validator: perform sync committee duties

### PR DESCRIPTION
**Pull request by terencechain**
# Description
Part of #9123 
Validator can submit 1.) sync committee message 2.) sync committee contribution to the beacon node

### CODE REVIEW

File: validator/client/sync_committee.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+	for i, index := range indexRes.Indices {
+		subCommitteeSize := params.BeaconConfig().SyncCommitteeSize / params.BeaconConfig().SyncCommitteeSubnetCount
+		subnet := uint64(index) / subCommitteeSize
+		selectionProof, err := v.signSyncSelectionData(ctx, pubKey, subnet, slot)
+		if err != nil {
+			log.Errorf("Could not sign selection data: %v", err)
+			return
+		}
+		selectionProofs[i] = selectionProof
+	}
```
Comment (rauljordan) [inline validator/client/sync_committee.go:115]: This section is too big, should be split up and tested

File: validator/client/sync_committee.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		if err != nil {
+			log.Errorf("Could check in aggregator: %v", err)
+			return
+		}
+		if !isAggregator {
+			continue
+		}
+		subCommitteeSize := params.BeaconConfig().SyncCommitteeSize / params.BeaconConfig().SyncCommitteeSubnetCount
+		subnet := uint64(comIdx) / subCommitteeSize
+		contribution, err := v.validatorClient.GetSyncCommitteeContribution(ctx, &ethpb.SyncCommitteeContributionRequest{
```
Comment (rauljordan) [inline validator/client/sync_committee.go:130]: How many requests will we be making on average here? Maybe in the future could involved some batching

Comment (terencechain) [reply validator/client/sync_committee.go:130]: This is a very rare event, I rather keep it simple.

64 out of N on per slot base. N is the total validator count

File: validator/client/sync_committee.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+	sig, err = v.keyManager.Sign(ctx, &validatorpb.SignRequest{
+		PublicKey:       pubKey[:],
+		SigningRoot:     root[:],
+		SignatureDomain: d.SignatureDomain,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return sig.Marshal(), nil
```
Comment (rauljordan) [inline validator/client/sync_committee.go:236]: This logic is duplicated

**Review by rauljordan — APPROVED** [review]
Thank you, this is now very clear!

**Review by rauljordan — APPROVED** [review]

**Review by rauljordan — APPROVED** [review]

**Pull request merged**